### DISCUSSION TITLE
Why can't I see the sphere？

### DISCUSSION THREAD

**Discussion by duanlinlim** [InteractiveComputerGraphics/SPlisHSPlasH#343] [General]
{
	"Configuration": 
	{
		"timeStepSize": 0.001,
		"numberOfStepsPerRenderUpdate": 2,
		"particleRadius": 0.025, 
		"density0": 1000, 
		"simulationMethod": 4,
		"gravitation": [0,-9.81,0], 
		"cflMethod": 1, 
		"cflFactor": 0.5,
		"cflMaxTimeStepSize": 0.005,
		"maxIterations": 100,
		"maxError": 0.01,
		"maxIterationsV": 100,
		"maxErrorV": 0.1,		
		"stiffness": 50000,
		"exponent": 7,
		"velocityUpdateMethod": 0,
		"enableDivergenceSolver": true,
		"boundaryHandlingMethod": 2
	},
	"Materials": [
		{
			"id": "Fluid",
			"viscosity": 0.01,
			"viscosityMethod": 1,
			"colorMapType": 1
		}
	],
	"Simulation": 
	{
		"timeStepSize": 0.005,
		"maxIter" : 5,
		"maxIterVel" : 5,
		"velocityUpdateMethod" : 0,
		"contactTolerance": 0.05,
		"tetModelSimulationMethod": 2,
		"triangleModelSimulationMethod": 2,
		"triangleModelBendingMethod": 2,
		"contactStiffnessRigidBody" : 1.0,
		"contactStiffnessParticleRigidBody": 100.0,
		"cloth_stiffness": 1.0,
		"cloth_bendingStiffness": 0.005,
		"cloth_xxStiffness": 1.0,
		"cloth_yyStiffness": 1.0,
		"cloth_xyStiffness": 1.0,
		"cloth_xyPoissonRatio": 0.3,
		"cloth_yxPoissonRatio": 0.3,
		"cloth_normalizeStretch": 0,
		"cloth_normalizeShear": 0, 
		"solid_stiffness": 1.0,
		"solid_poissonRatio": 0.2,
		"solid_normalizeStretch": 0,
		"solid_normalizeShear": 0
	},		
	"RigidBodies": [
		{
			"id": 1,
			"geometryFile": "../models/UnitBox.obj",
			"translation": [0,1.5,0],
			"rotationAxis": [1, 0, 0],
			"rotationAngle": 0,
			"scale": [3.1, 3.1, 3.1],
			"color": [0.5, 0.5, 0.5, 1.0], 
			"isDynamic": false,
			"restitution" : 0.6,
			"friction" : 0.0,
			"collisionObjectType": 2,
			"collisionObjectScale": [3.1, 3, 3.1],	
			"testMesh": 1, 
			"isWall": true,
			"invertSDF": true,
			"mapInvert": true, 
			"mapThickness": 0.0,
			"mapResolution": [30,30,30]
		},
		{
			"id": 2,
			"geometryFile": "../models/sphere.obj",
			"particleFile": "../models/sphere.bgeo",
			"isDynamic": true, 
			"density": 20, 
			"translation": [0.0,1.0,0],
			"rotationAxis": [1, 0, 0],
			"rotationAngle": 0.0,
			"scale": [0.5, 0.5, 0.5],
			"velocity": [1,0,0],
			"restitution" : 0.6,
			"friction" : 0.2,
			"color": [0.3, 0.5, 0.8, 1.0], 
			"collisionObjectType": 1,
			"collisionObjectScale": [0.5, 0.5, 0.5], 
			"testMesh": 1, 
			"isWall": false,
			"mapInvert": false, 
			"mapThickness": 0.0,
			"mapResolution": [25,25,25]
		}
	],
	"FluidBlocks": [
		{
			"denseMode": 0,
			"start": [-1.5, 0.0, -1.5],
			"end": [-0.8, 0.75, -0.8]
		}, 
		{
			"denseMode": 0,
			"start": [0.8, 0.0, 0.8],
			"end": [1.5, 2.75, 1.5]
		}Why can't I see two balls when simulating this JSON file? Is it because the color of the balls is the same as that of the fluid? How should I modify it?
	]

**Comment by janbender** [comment]
The json file defines just one sphere. Why you expect to see two?

**Reply by duanlinlim** [reply]
sorry 
The JSON file is this
{
	"Configuration": 
	{
		"pause": true,
		"sim2D": true, 
		"timeStepSize": 0.001,
		"particleRadius": 0.025,
		"colorMapType": 1,
		"numberOfStepsPerRenderUpdate": 4,
		"density0": 1000, 
		"simulationMethod": 4,
		"gravitation": [0,-981,0], 
		"cflMethod": 1, 
		"cflFactor": 1,
		"cflMaxTimeStepSize": 0.005,
		"maxIterations": 100,
		"maxError": 0.1,
		"maxIterationsV": 100,
		"maxErrorV": 0.1,		
		"stiffness": 50000,
		"exponent": 7,
		"velocityUpdateMethod": 0,
		"enableDivergenceSolver": true,
		"boundaryHandlingMethod": 2,
		"enablePartioExport": true,
		"particleAttributes": "density",
		"partioAttributes": "density",
		"enableVTKExport": true
		
	},
	"Materials": [
		{
			"id": "Fluid",
			"surfaceTension": 0.2,
			"surfaceTensionMethod": 0,		
			"viscosity": 0.01,
			"viscosityMethod": 1, 
			"vorticityMethod": 1, 
			"vorticity": 0.1, 
			"viscosityOmega": 0.05,
			"inertiaInverse": 0.5,
			"renderMinValue": 0.0,
			"renderMaxValue": 5.0,
			"colorMapType": 4
		}
	],
	"RigidBodies": [
		{
			"id": 1,
			"geometryFile": "../models/UnitBox.obj",
			"translation": [0,3,0],
			"rotationAxis": [1, 0, 0],
			"rotationAngle": 0,
			"scale": [2, 6, 2.0],
			"color": [0.5, 0.5, 0.5, 1.0], 
			"isDynamic": false,
			"restitution" : 0.6,
			"friction" : 0.0,
			"collisionObjectType": 2,
			"collisionObjectScale": [3.1, 3, 3.1],	
			"testMesh": 1, 
			"isWall": true,
			"invertSDF": true,
			"mapInvert": true, 
			"mapThickness": 0.0,
			"mapResolution": [30,30,30]
		},
		{
			"id": 2,
			"geometryFile": "../models/sphere.obj",
			"isDynamic": false, 
			"density": 1500, 
			"translation": [1.0,5,0],
			"rotationAxis": [1, 0, 0],
			"rotationAngle": 0.0,
			"scale": [0.5, 0.5, 0.5],			
			"restitution" : 0.6,
			"friction" : 0.2,
			"color": [0.3, 0.5, 0.8, 1.0], 
			"collisionObjectType": 1,
			"collisionObjectScale": [0.5, 0.5, 0.5], 
			"testMesh": 1, 
			"isWall": false,
			"mapInvert": false, 
			"mapThickness": 0.0,
			"mapResolution": [25,25,25]
		},
		{
			"id": 3,
			"geometryFile": "../models/sphere.obj",
			"isDynamic": false, 
			"density": 1500, 
			"translation": [1.0,4.5,0],
			"rotationAxis": [1, 0, 0],
			"rotationAngle": 0.0,
			"scale": [0.5, 0.5, 0.5],
			"restitution" : 0.6,
			"friction" : 0.2,
			"color": [0.3, 0.5, 0.8, 1.0], 
			"collisionObjectType": 1,
			"collisionObjectScale": [0.5, 0.5, 0.5], 
			"testMesh": 1, 
			"isWall": false,
			"mapInvert": false, 
			"mapThickness": 0.0,
			"mapResolution": [25,25,25]
		}
	],
	"FluidBlocks": [
		{
			"denseMode": 0,
			"start": [-1, 0.0, -1],
			"end": [1, 6.0, 1],
			"translation": [0.025, 0.025, 0],
			"scale": [1, 1, 1]
		}
	]	
}

**Reply by janbender** [reply]
When I load your scene, there are two spheres.

**Reply by duanlinlim** [reply]
Can the color of particles be changed? Is it because fluid particles and spheres have the same color, causing the two spheres to not be seen
The simulated GUI interface is gray white with no color？
The simulation result shows that the particles are jumping and there is no evidence of two small balls falling and deviating？

**Reply by janbender** [reply]
You can change the color of the rigid bodies or the color scheme of the particles. 

The simulation tends to get unstable since you are using a very large gravitation acceleration (100x larger than usual). To do a simulation which such accelerations you will need other tolerances for the solver and far smaller time steps.

**Reply by duanlinlim** [reply]
The color scheme of rigid bodies or particles. Does it refer to the parameter color

**Reply by duanlinlim** [reply]
Is sphere. obj in this model a radius of 1
